### PR TITLE
Adding new keyboard: Trkeyboards trk1

### PR DESCRIPTION
**Pull request by TvrdRadKeyboards**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/trkeyboards/trk1/config.h [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+#define ENCODERS_PAD_B { B0 }
+#define UNUSED_PINS
+
+/* COL2ROW, ROW2COL */
+#define DIODE_DIRECTION COL2ROW
+
+/* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
+#define DEBOUNCE 5
+#define LOCKING_SUPPORT_ENABLE
+#define LOCKING_RESYNC_ENABLE
```
Comment (drashna) [inline keyboards/trkeyboards/trk1/config.h:44]:
```suggestion
```

File: keyboards/trkeyboards/trk1/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    if (index == 0) {
+        if (clockwise) {
+            tap_code_delay(KC_VOLU, 10);
+        } else {
+            tap_code_delay(KC_VOLD, 10);
+        }
+    }
+    return true;
+}
+#endif
```
Comment (drashna) [inline keyboards/trkeyboards/trk1/keymaps/default/keymap.c:63]:
```suggestion
```

File: keyboards/trkeyboards/trk1/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1 @@
+#include "../default/keymap.c"
```
Comment (drashna) [inline keyboards/trkeyboards/trk1/keymaps/via/keymap.c:1]: Please include the full keymap.

File: keyboards/trkeyboards/trk1/readme.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
+# Tvrd Rad Keyboards TRK1
+
+![TRK1](https://i.imgur.com/nTHrBMr.jpg)
```
Comment (drashna) [inline keyboards/trkeyboards/trk1/readme.md:3]:
```suggestion
![TRK1](https://i.imgur.com/nTHrBMrh.jpg)
```

File: keyboards/trkeyboards/trk1/readme.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+*A short description of the keyboard/project*
```
Comment (drashna) [inline keyboards/trkeyboards/trk1/readme.md:5]: ???

File: keyboards/trkeyboards/trk1/readme.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+* Keyboard Maintainer: [Tvrd Rad Keyboards](https://www.instagram.com/trkeyboards/)
+* Hardware Supported: Pro Micro and clones
+* Hardware Availability: https://www.instagram.com/trkeyboards/
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/trkeyboards/trk1/readme.md:11]: Could you add instructions on how to enter the bootloader, to flash new firmware? 

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```

File: keyboards/trkeyboards/trk1/rules.mk [hunk 7]
```diff
@@ -0,0 +1,32 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
```
Comment (drashna) [inline keyboards/trkeyboards/trk1/rules.mk:10]: There is no "full" bootmagic anymore, only the "lite" option, which can be turned on with just "yes".
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

File: keyboards/trkeyboards/trk1/rules.mk [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
```
Comment (drashna) [inline keyboards/trkeyboards/trk1/rules.mk:18]:
```suggestion
NKRO_ENABLE = yes           # Enable N-Key Rollover
```

File: keyboards/trkeyboards/trk1/rules.mk [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (drashna) [inline keyboards/trkeyboards/trk1/rules.mk:21]:
```suggestion
```

File: keyboards/trkeyboards/trk1/rules.mk [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+AUDIO_ENABLE = no           # Audio output
+ENCODER_ENABLE = yes
+# OLED - #OLED_DRIVER_ENABLE = yes
```
Comment (drashna) [inline keyboards/trkeyboards/trk1/rules.mk:24]:
```suggestion
```

File: keyboards/trkeyboards/trk1/rules.mk [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+
+# Firmware size reductions
+GRAVE_ESC_ENABLE = no
+MAGIC_ENABLE = no
+SPACE_CADET_ENABLE = no
+ifneq ($(PLATFORM),CHIBIOS)
+    LTO_ENABLE = yes
+endif
```
Comment (drashna) [inline keyboards/trkeyboards/trk1/rules.mk:32]: This should be moved to `post_rules.mk`

File: keyboards/trkeyboards/trk1/trk1.c [hunk 12]
```diff
@@ -0,0 +1,17 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "trk1.h"
```
Comment (drashna) [inline keyboards/trkeyboards/trk1/trk1.c:17]: This allows for a default behavior to be defined, without the need for the user to specify the behavior.  This is very useful for the configurator which doesn't have a way to configure this behavior, currently.

```suggestion
#include "trk1.h"

#ifdef ENCODER_ENABLE
bool encoder_update_kb(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index, clockwise)) { return false; }
    if (index == 0) {
        if (clockwise) {
            tap_code_delay(KC_VOLU, 10);
        } else {
            tap_code_delay(KC_VOLD, 10);
        }
    }
    return true;
}
#endif
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/trkeyboards/trk1/config.h [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+#define FORCE_NKRO
+#define USB_POLLING_INTERVAL_MS 1
+
+/* Tapping */
+#define PERMISSIVE_HOLD
+#define TAPPING_TERM   200
+#define TAPPING_TOGGLE 2
+
+/* Firmware size reductions */
+#undef  LOCKING_SUPPORT_ENABLE
```
[the remark this answers is not in the document]
Comment (vomindoraan) [reply keyboards/trkeyboards/trk1/config.h:56]: Why the removal?

**Pull request closed without merging**